cmd/mdatagen/metricdata.go

```diff
@@ -149,9 +158,9 @@ type sum struct {
 
 // Unmarshal is a custom unmarshaler for sum. Needed mostly to avoid MetricValueType.Unmarshal inheritance.
 func (d *sum) Unmarshal(parser *confmap.Conf) error {
-	if !parser.IsSet("aggregation_temporality") {
-		return errors.New("missing required field: `aggregation_temporality`")
-	}
+	// if !parser.IsSet("aggregation_temporality") {
+	// 	return errors.New("missing required field: `aggregation_temporality`")
+	// }
```

Comment:
Can this be deleted?

Reply:
It can, i was leaving it in here to remind myself to talk about it w/ @dmitryax :D

Reply:
Deleted

Reply:
We can probably move this validation level app and apply it differently to `metrics` (require) and `internal_metrics` (require no be not set).

Reply:
I tried to do this for a bit, but the problem is that since `metrics` are being unmarshaled as a map, there isn't a good place to inject the logic to return an error only when parsing `metrics` and not `telemetry:metrics`

Will open an issue to ensure this functionality can be brought back later on